tap_s3_csv/__init__.py

```diff
@@ -99,6 +100,9 @@ def main():
                 break
         except BaseException as err:
             LOGGER.error(err)
+        
+        # If not external source, it is from importing csv (replacement for tap-csv) 
+        dialect.detect_tables_dialect(config)
```

Comment:
why don't we need to detect dialect if it's external?

Reply:
We only have external_id in config for s3 connectors. For s3 connectors, it had a config modal in UI where user could select the delimiter to use. So the original behaviour for s3 connectors was to use the delimiter selected by user along with default quotechar and utf-8 encoding whereas for csv imports, we just upload the csv file and detect dialect in tap. I tried not to change the original behaviour, and also, it was ambiguous which dialect to use when detected delimiter differed from the one passed in by user. Should we still detect dialects for s3 connectors?

Reply:
hmm no but maybe we should have originally detected. it's fine, leave it the way it is.